File: TwitterDealer/TwitterDealer/Models/TwitterUserModels/StatusTweet.cs
using System;
using System.Collections.Generic;
using TwitterDealer.Models.BaseModels;

namespace TwitterDealer.Models.TwitterUserModels
{
	public class StatusTweet
	{
		public bool IsFavourite { get; set; }

		public int RetweetCount { get; set; }

		public string TweetText { get; set; }

		public string Language { get; set; }

		public IEnumerable<BaseUserMedia> MediaUrl { get; set; }

		public bool? IsPossiblySensitive { get; set; }

		public string Url { get; set; }

		public DateTime? Created { get; set; }

		public IEnumerable<StatusTweet> Replies { get; set; }
	}
}

File: TwitterDealer/TwitterDealer.WebClientApp/ClientApp/src/app/models/statusTweet.ts
import { BaseUserMedia } from './baseUserMedia';

export interface StatusTweet {
    isFavourite: boolean;
    retweetCount: number;
    tweetText: string;
    language: string;
    mediaUrl: BaseUserMedia[];
    isPossiblySensitive?: boolean;
    url: string;
    favoriteCount: number;
    created?: Date;
    userName: string;
    userScreenName: string;
    replies: StatusTweet[];
}
